source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -149,32 +152,27 @@ bool OsSysCallsImpl::supportsIpTransparent() const {
   // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/net/ipv6/ipv6_sockglue.c?id=6c46862280c5f55eda7750391bc65cd7e08c7535
   //
   // So, almost recent linux kernel supports both IP_TRANSPARENT and IPV6_TRANSPARENT options.
+  // But there are also has ipv4 only or ipv6 only scenarios.
   //
   // And these socket options need CAP_NET_ADMIN capability to be applied.
   // The CAP_NET_ADMIN capability should be applied by root user before call this function.
-  static const bool is_supported = [] {
-    // Check ipv4 case
-    int fd = ::socket(AF_INET, SOCK_DGRAM | SOCK_NONBLOCK, IPPROTO_UDP);
+
+  static constexpr auto transparent_supported = [](int family) {
+    auto opt_tp = family == AF_INET ? ENVOY_SOCKET_IP_TRANSPARENT : ENVOY_SOCKET_IPV6_TRANSPARENT;
```

Comment:
nit, we can just use `Network::Address::IpVersion` directly here instead of using AF_INET

Reply:
Thanks for comment. The macro definitions have different values, which means they cannot be used directly. Such as, the `Network::Address::IpVersion::v4` has a value of 0, `AF_INET` has a value of 2.

https://github.com/envoyproxy/envoy/blob/aa5ab8d093996d9c141c9f5f68abecbc79e7183a/envoy/network/address.h#L74-L76

[Linux socket](https://students.mimuw.edu.pl/SO/Linux/Kod/include/linux/socket.h.html)

```
/* Supported address families. */
#define AF_UNSPEC	0
#define AF_UNIX		1	/* Unix domain sockets 		*/
#define AF_INET		2	/* Internet IP Protocol 	*/
#define AF_AX25		3	/* Amateur Radio AX.25 		*/
#define AF_IPX		4	/* Novell IPX 			*/
#define AF_APPLETALK	5	/* Appletalk DDP 		*/
#define	AF_NETROM	6	/* Amateur radio NetROM 	*/
#define AF_BRIDGE	7	/* Multiprotocol bridge 	*/
#define AF_AAL5		8	/* Reserved for Werner's ATM 	*/
#define AF_X25		9	/* Reserved for X.25 project 	*/
#define AF_INET6	10	/* IP version 6			*/
#define AF_MAX		12	/* For now.. */

```

Reply:
oh, sorry, I didn't say it clearly. I mean it can be written like this:
```
static constexpr auto transparent_supported = [](Network::Address::IpVersion version) {
    auto opt_tp = version== Network::Address::IpVersion:v4? ENVOY_SOCKET_IP_TRANSPARENT : ENVOY_SOCKET_IPV6_TRANSPARENT;
```

then you needn't convert between Network::Address::IpVersion and AF_INET

But yes, I just saw, you need the family later when calling the `socket()`. so yes, the current code is totally fine.

Reply:
yeah.